specs/dwritecore/DWriteCore.md

```diff
@@ -510,9 +510,217 @@ The following other DirectWrite APIs also use the hybrid font selection algorith
+`DrawGlyphRun` methods that render glyph runs by filling the glyph shapes with a specified text
+color. For convenience, we will refer to this as _monochrome_ glyph rendering. All fonts have
+monochrome glyphs, but a color font also has color representations of some glyphs. To render glyphs
+in color, an application must use different glyph rendering APIs instead of calling the monochrome
+`DrawGlyphRun` method.
+
+Because color font support is implemented at the glyph rendering level, it does not affect text
+layout. This is true whether a client uses the `IDWriteTextLayout` interface or implements its own
+text layout algorithm. The mapping of characters to glyphs and the positioning of those glyphs all
+use monochrome glyph IDs and their associated metrics. The output of the text layout process is a
+sequence of monochrome glyph runs. Color font support can be enabled by translating those monochrome
+"base" glyph runs to color glyphs runs at rendering time.
+
+The easiest way to enable color is to call the `DrawGlyphRunWithColor` method instead of calling
+`DrawGlyphRun`. Direct2D exposes this method through the `ID2D1DeviceContext7` interface.
+DirectWrite exposes this method through the `IDWriteBitmapRenderTarget3` interface.
+
+At a lower level, enabling color involves translating the monochrome _base glyph run_ to a sequence
+of _color glyph runs_. Each color glyph run must then be rendered using the appropriate method,
+depending on its _glyph image format_. The translation is performed by the factory object's
+`TranslateColorGlyphRun` method. It and related APIs are described in the following sections.
+
+### Using TranslateColorGlyphRun
+
+The `TranslateColorGlyphRun` factory method translates a monochrome _base glyph run_ into a sequence
+of _color glyph runs_. There are multiple ways of representing color glyphs, known as _glyph image
+formats_ and identified by the `DWRITE_GLYPH_IMAGE_FORMATS` enumeration. When calling
+`TranslateColorGlyphRun`, the client specifies the combination of glyph image formats it supports.
+The `TranslateColorGlyphRun` returns a _glyph run enumerator_ object
+(`IDWriteColorGlyphRunEnumerator1` interface), which the client uses to iterate over the sequence of
+color glyph runs, represented by the `DWRITE_COLOR_GLYPH_RUN1` structure. Each color glyph run has a
+glyph image format. The client branches on the color run's glyph image format to determine how to
+render the color run.
+
+If the base glyph run does not have color glyphs in any of the requested glyph image formats, the
+`TranslateColorGlyphRun` method returns `DWRITE_E_NOCOLOR`, in which case the client should render
+the base glyph run as a monochrome glyph run.
+
+Otherwise, the base glyph run may contain a mix of color glyphs in different glyph image formats as
+well as monochrome glyphs. For each base glyph, the `TranslateColorGlyphRun` method determines the
```

Comment:
What happens if the font uses a glyph image format that the caller doesn't support? Do we auto-convert it to something the caller *does* support? I think what happens is that we look for fallback formats that the caller supports, and if no fallback can be found, we render as monochrome.

Reply:
You think correctly. We don't convert, but we only return formats the client supports with monochrome being the ultimate fallback.